server/portal/apps/onboarding/api/views.py

```diff
@@ -223,14 +251,18 @@ def post(self, request, username):
         # Call action handler
         if action == "reset":
             self.reset(request, setup_step)
+            execute_setup_steps.apply_async(args=[user.username])
+        elif action == "force_create_storage":
+            self.force_create_storage(request, setup_step)
         elif action == "complete":
             self.complete(request, setup_step)
+            execute_setup_steps.apply_async(args=[user.username])
         else:
             self.client_action(request, setup_step, action, data)
+            execute_setup_steps.apply_async(args=[user.username])
 
         # If no exception was generated from any of the above actions, continue.
         # Retry executing the setup queue for this user
-        execute_setup_steps.apply_async(args=[user.username])
```

Comment:
I see this is moved into the if/else block for everything except `force_create_storage`.  

but the comments explaining it remain here.  Could this be refactored so that `execute_setup_steps.apply_async(args=[user.username])` stays here (instead of in if/else blocks).  but look something like:

```
if action != `force_create_storage:`
    execute_setup_steps.apply_async(args=[user.username])
```